batch/sql/estimated-current.txt

```diff
@@ -186,14 +197,15 @@ BEGIN
   END IF;
 END $$
 
+DROP TRIGGER IF EXISTS attempts_before_update;
 CREATE TRIGGER attempts_before_update BEFORE UPDATE ON attempts
 FOR EACH ROW
 BEGIN
   IF OLD.start_time IS NOT NULL AND (NEW.start_time IS NULL OR NEW.start_time < OLD.start_time) THEN
     SET NEW.start_time = OLD.start_time;
   END IF;
 
-  IF OLD.end_time IS NOT NULL AND (NEW.end_time IS NULL OR NEW.end_time > OLD.end_time) THEN
+  IF OLD.reason IS NOT NULL AND (NEW.end_time IS NULL OR NEW.end_time > OLD.end_time) THEN
```

Comment:
This is quite hard to think about.  Thinking out load:

First, I'm assuming if end_time IS NOT NULL, then reason IS NOT NULL: if we set the end time, it must have been for a reason.

We only want to take the newer value if there is no old reason, or the newer time is < older time (and both are not null).  Formally, take new if:

 > OLD.reason IS NULL OR (OLD.end_time IS NOT NULL AND NEW.end_time IS NOT NULL AND NEW.end_time < OLD.end_time)

Conversely, taking the opposite, we get:

 > OLD.reason IS NOT NULL AND (OLD.end_time IS NULL OR NEW.end_time IS NULL OR NEW.end_time >= OLD.end_time)

So I think you need a `OLD.end_time IS NULL` in the condition, too.  In particular, I think that's necessary to fix the example you had.  I'd also change it to keep the old one on ties, but meh, whatever.